app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -196,13 +212,47 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote): Musi
 		}
 	}
 
+	/**
+	 * Retrieves the cover art Bitmap for the provided ImageUri. If it is present in the coverArtCache
+	 * then the Bitmap is returned otherwise the imagesApi is called to asynchronously add the cover
+	 * art Bitmap to the cache and returns null in the meantime.
+	 */
+	fun getCoverArt(imageUri: ImageUri): Bitmap? {
+		val coverArt = coverArtCache.get(imageUri)
+		if (coverArt != null) {
+			return coverArt
+		} else {
+			remote.imagesApi.getImage(imageUri, Image.Dimension.THUMBNAIL).setResultCallback { coverArtResponse ->
+				coverArtCache.put(imageUri, coverArtResponse)
+			}
+		}
+		return null
+	}
+
+	/**
+	 * Creates the QueueMetadata for the current queue containing the title, subtitle, queue songs,
+	 * and queue cover art.
+	 */
+	private fun buildQueueMetadata() {
+		val recentlyPlayedUri = "com.spotify.recently-played"
```

Comment:
Oh interesting, I hadn't thought of this! Does the contextUri not provide access to the cover art?
I could see value in fetching more than just the first recently-played item, in case some interaction between other clients (Web, Sonos, who knows) pushes more items on this recently-played list than what the phone has currently queued up.

Reply:
Yeah it's kind of hacky but it was the only way I could think of to get the playlist/queue cover art. I was hoping there was a more native way of doing it but after realizing there wasn't had to get a little creative :). When I was testing different ways of getting the current playlist/queue cover art I wasn't able to get it working through the contextUri though. The one thing I made sure to test when I was using this method was setting the different playlists through the web player, desktop player and phone player and so far the recently played list is very reliable and near instantaneously updated across all instances which makes it great for this purpose